File: move_zeroes/MoveZeroes.java
class MoveZeroes {
    public void moveZeroes(int[] nums) {
        if (nums == null || nums.length < 2) {
            return;
        }
        
        int slow = 0; 
        for (int fast = 0; fast < nums.length; fast++) {
            if (nums[fast] == 0) {
                continue;
            }
            nums[slow++] = nums[fast];
        }
        
        for (; slow < nums.length; slow++) {
            nums[slow] = 0;
        }
    }
}

File: move_zeroes/Move_Zeroes.cpp
class Move_Zeroes {
public:
    void moveZeroes(vector<int>& nums) {
        if (nums.empty() || nums.size() < 2) {
            return;
        }
        
        int ptr = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != 0) {
                nums[ptr++] = nums[i];
            }
        }
        for (; ptr < nums.size(); ptr++) {
            nums[ptr] = 0;
        }
    }
};